src/resolvers/ResolverGenerator.ts

```diff
@@ -0,0 +1,68 @@
+import { readFileSync } from 'fs';
+import { GlobSync } from 'glob';
+import * as handlebars from 'handlebars';
+import { basename, join } from 'path';
+import { ResolverInstance } from '../resolvers/ResolverInstance';
+import { HandlebarsHelpers } from '../schema/Helpers';
+
+/**
+ * Generate Typescript compatible schema
+ */
+export class GraphQLResolverGenerator {
+  private rootTemplate: string = `{{> resolvers}}`
+
+  /**
+   * Register helper functions for Handlebars
+   */
+  constructor() {
+    new HandlebarsHelpers().registerHelpers()
+  }
+
+  /**
+   * Generate Typescript compatible schema
+   * @param resolvers Array of ResolverInstance
+   */
+  public generateResolvers(resolvers: ResolverInstance[]): string {
+    this.registerPartials()
+
+    const template = handlebars.compile(this.rootTemplate)
+    
+    const input: any = {}
+    input.resolvers = this.splitByTypes(resolvers)
+
+    return template(input)
+  }
+
+  /**
+   * Split resolvers into Query and Mutations
+   * @param resolvers Array of ResolverInstance
+   */
+  private splitByTypes(resolvers: ResolverInstance[]) {
+    const computedResolvers = { query: [], mutation: [] };
+    computedResolvers.mutation = resolvers.filter((value: ResolverInstance) => {
+      return value.resolverType === "Mutation"
+    })
+    computedResolvers.query = resolvers.filter((value: ResolverInstance) => {
```

Comment:
See my proposition above. Please do not feel attached to implementation that was made in 2 days under time pressure to demo this project to stackholders. Code here is really rough and it requires refactoring. Let's not base on this and adapt it to our needs.

Reply:
How do you propose we go with implementing the resolvers now?